platform/local/cluster.go

```diff
@@ -0,0 +1,146 @@
+/*
+   Copyright 2015 CoreOS, Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package local
+
+import (
+	"fmt"
+	"os/exec"
+	"runtime"
+
+	"github.com/coreos/coreos-cloudinit/config"
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+
+	"github.com/coreos/mantle/network"
+)
+
+type LocalCluster struct {
+	SSHAgent    *network.SSHAgent
+	Dnsmasq     *Dnsmasq
+	ConfigDrive *ConfigDrive
+	nshandle    netns.NsHandle
+}
+
+func NewLocalCluster() (*LocalCluster, error) {
+	// Before creating a new namespace lock to a thread and restore the
+	// original namespace on return to avoid confusing other goroutines.
+	runtime.LockOSThread()
+	defer runtime.UnlockOSThread()
+
+	origns, err := netns.Get()
+	if err != nil {
+		return nil, err
+	}
+	defer netns.Set(origns)
+
+	lc := &LocalCluster{}
+	lc.nshandle, err = netns.New()
+	if err != nil {
```

Comment:
Generally I didn't inline multi-value assignments in order to keep the usage of `=` and `:=` consistent.